NMS-13885: Tuning broker docs (#4075)
OpenNMS/opennms: docs/modules/reference/pages/configuration/tuning-kafka.adoc

* link:https://kafka.apache.org/10/documentation.html#producerconfigs[Producer Configs] for RPC communication
* link:https://kafka.apache.org/10/documentation.html#newconsumerconfigs[New Consumer Configs] for Sink communication

== Multi-tenancy with {page-component-title} and Kafka

Topics will be automatically created and are prefixed by default with `OpenNMS`.
If you want to use an Apache Kafka cluster with multiple {page-component-title} instances, customize the topic prefix by setting the `org.opennms.instance.id` system property on {page-component-title} and all its Minions and Sentinels to a string value that identifies your instance.

.`$\{OPENNMS_HOME}/etc/opennms.properties.d/instance-id.properties`
[source, properties]
----
org.opennms.instance.id=MyNMS
----

Update the Minion's instance ID accordingly to match the {page-component-title} instance.

.`$\{MINION_HOME}/etc/custom.system.properties`.
[source, properties]
----
org.opennms.instance.id=MyNMS
----

== Tips for Kafka

> * For Kafka RPC, the number of partitions should always be greater than the number of minions at a location.

Reviewer: ```suggestion
* For Kafka RPC, the number of partitions should always be greater than the number of Minions at a location.
```

When there are multiple locations, partitions >= max number of minions at a location.

> * By default, Kafka RPC supports buffers greater than >1MB by splitting large buffer into chunks of 900KB.

Reviewer: ```suggestion
* By default, Kafka RPC supports buffers greater than >1MB by splitting a large buffer into chunks of 900KB.
```

Author: ```suggestion
* By default, Kafka RPC supports buffers greater than >1MB by splitting them into chunks of 900KB.
```

> Max buffer size (900KB, by default) can be configured by setting `org.opennms.core.ipc.rpc.kafka.max.buffer.size` (in bytes).

Reviewer: ```suggestion
Configure max buffer size (900KB, by default) by setting `org.opennms.core.ipc.rpc.kafka.max.buffer.size` (in bytes).
```

Author: ```suggestion
Configure max buffer size (900KB, by default) by setting `org.opennms.core.ipc.rpc.kafka.max.buffer.size` to the desired size in bytes.
```

* Default time to live (time at which request will expire) is 20000 msec (20sec).

> It can be changed by configuring system property `org.opennms.core.ipc.rpc.kafka.ttl` in msec.

Reviewer: ```suggestion
To change it, configure the system property `org.opennms.core.ipc.rpc.kafka.ttl` in msec.
```
